lib/tasks/plugin_test.rake

```diff
@@ -0,0 +1,79 @@
+require File.expand_path('../../config/environment', File.dirname(__FILE__))
+
+desc 'Run a single or multiple plugins tests - called via rake plugin:test[plugin_name,plugin_name]'
+task 'plugin:test' => [:environment] do |t, args|
+  plugin_names = args.extras
+    
+  puts "You must specify the name of at least one plugin (e.g. rake test:plugin['my_plugin'])" if plugin_names.empty?
+
+  plugins = plugin_names.collect do |plugin_name|
+    plugin = Foreman::Plugin.find(plugin_name)
+    unless plugin
+      puts "No plugin found for #{plugin_name}. Availble plugins are:"
+      Rake::Task['plugins'].invoke
+      exit 1
+    end
+    plugin
+  end
+
+  plugins.each do |plugin|
+    Rake::Task["test:units"].enhance do
+      test_task = plugin.test_task.nil? ? "plugin:test:#{plugin.name}" : plugin.test_task
+      Rake::Task[test_task].invoke
+    end
+  end
+
+  Rake::Task['test:units'].invoke
+end
+
+desc 'CI plugin test task for running a single or multiple plugins tests - called via rake plugin:jenkins[plugin_name,plugin_name]'
+task 'plugin:jenkins' => [:environment] do |t, args|
+  plugin_names = args.extras
+    
+  puts "You must specify the name of at least one plugin (e.g. rake test:plugin['my_plugin'])" if plugin_names.empty?
+
+  # Reset to the base Jenkins unit test in case plugins are enhancing it
+  Rake::Task['jenkins:unit'].clear
+  load 'lib/tasks/jenkins.rake'
+
+  plugins = plugin_names.collect do |plugin_name|
+    plugin = Foreman::Plugin.find(plugin_name)
+    unless plugin
+      puts "No plugin found for #{plugin_name}. Availble plugins are:"
+      Rake::Task['plugins'].invoke
+      exit 1
+    end
+    plugin
+  end
+
+  plugins.each do |plugin|
+    Rake::Task["jenkins:unit"].enhance do
+      test_task = plugin.test_task.nil? ? "plugin:test:#{plugin.name}" : plugin.test_task
+      Rake::Task[test_task].invoke
```

Comment:
Perhaps do:

    test_task = plugin.test_task.nil? ? "plugin:test:#{plugin.name}" : plugin.test_task
    Rake::Task["jenkins:unit"].enhance test_task

This would add the plugin task as a pre-req for jenkins:unit, in the same way that jenkins.rake works, rather than putting it in the body of the task.